### PR TITLE
Add pyquil

### PR DESCRIPTION
**Pull request by mpharrigan**

### CODE REVIEW

File: recipes/pyquil/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  run:
+    - python
+    - numpy
+    - six
+    - antlr-python-runtime
+    - requests
+
+test:
+  imports:
```
[the remark this answers is not in the document]
Comment (mpharrigan) [reply recipes/pyquil/meta.yaml]: Unfortunately the CLI entry point is just a little script that write as config file to `~/.pyquil_config`. It doesn't have a help flag and expects input

**Comment by mpharrigan** [comment]
Made the requested changes minus the cli test, see above

**Comment by mpharrigan** [comment]
bump. Do you know how to solve the windows errors?

**Comment by mpharrigan** [comment]
I was able to build the package on a local windows box without seeing the appveyor error

**Comment by carlodri** [comment]
> bump. Do you know how to solve the windows errors?

probably related to [this issue](https://github.com/conda/conda-build/issues/2627)

removing the `noarch` could fix things.

**Comment by CJ-Wright** [comment]
If this is noarch then we shouldn't need to fix the Windows error since all OSs will use the single noarch package.

File: recipes/pyquil/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+  # The pyQuil setup.py includes superfluous requirements
+  patches: remove-setup-requirements.patch
+
+build:
+  noarch: python
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
```
[the remark this answers is not in the document]
Comment (CJ-Wright) [reply recipes/pyquil/meta.yaml]: `setuptools` doesn't interact with `noarch` well and those docs are superseded by these: https://conda-forge.org/docs/meta.html#use-pip

[the remark this answers is not in the document]
Comment (carlodri) [reply recipes/pyquil/meta.yaml]: ok, I wasn't aware of that, thank you for your help!

File: recipes/pyquil/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+test:
+  imports:
+    - pyquil.api
+    - pyquil.gates
+    - pyquil.quil
+
+about:
+  home: http://forest.rigetti.com
+  license: Apache-2.0
+  license_family: Apache
```
Comment (CJ-Wright) [inline recipes/pyquil/meta.yaml]: This needs a license file.

File: recipes/pyquil/meta.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (CJ-Wright) [reply recipes/pyquil/meta.yaml]: Its a bug in our docs, PR here: https://github.com/conda-forge/staged-recipes/pull/5548#issuecomment-378604197

**Comment by mpharrigan** [comment]
Added the license file

**Comment by mpharrigan** [comment]
Ping :)

**Comment by CJ-Wright** [comment]
This seems reasonable to me, and as noarch only needs to pass on circle.

**Comment by CJ-Wright** [comment]
Thank you!

**Comment by mpharrigan** [comment]
Thanks for all your help!

**Pull request merged**